> LGTM. But please follow `mypy --strict` as you can. All function and its arguments should be annotated. Also, I found some subscriptable types is lacking of type arguments.

I added the missing type annotations in a new commit.
These two places required workarounds with empty sets:
`wbsv/crawler.py:56: error: Argument 1 to "union" of "set" has incompatible type "*List[Set[str]]"; expected "Iterable[<nothing>]"
wbsv/main.py:101: error: Argument 1 to "union" of "set" has incompatible type "*List[Set[str]]"; expected "Iterable[<nothing>]"
`
Related mypy issues:
https://github.com/python/mypy/issues/8087
https://github.com/python/mypy/issues/2013

--- a/test_wbsv.py
+++ b/test_wbsv.py
@@ -3,7 +3,7 @@ from urllib.parse import urlparse
 from wbsv import crawler, main
 
 
-#UrlFilter unit tests
+# UrlFilter unit tests
 from wbsv.url_filters import SchemaFilter, OwnDomainFilter
 
 
@@ -30,14 +30,19 @@ def test_own_domain_filter_should_reject_url_with_invalid_domain():
 
 
 def test_own_domain_filter_should_accept_url_with_valid_domain():
-    own_domain_filter =  OwnDomainFilter.from_string_urls(["https://example.com"])
+    own_domain_filter = OwnDomainFilter.from_string_urls(["https://example.com"])
     invalid_domain_url = urlparse("https://example.com")
 
     assert own_domain_filter.test_url(invalid_domain_url) is True
 
 
 def test_api_without_parser():
-    args = {"urls": ["https://example.com"], "own": False, "only_target": False, "level": 1}
+    args = {
+        "urls": ["https://example.com"],
+        "own": False,
+        "only_target": False,
+        "level": 1,
+    }
     c = crawler.Crawler.from_args(**args)
     urls = {"https://example.com", "https://www.iana.org/domains/example"}
     t = set().union(*c.run_crawl())
@@ -52,4 +57,3 @@ def test1():
     t = set().union(*c.run_crawl())
     if urls != t:
         raise AssertionError
-

--- a/wbsv/archiver.py
+++ b/wbsv/archiver.py
@@ -2,7 +2,7 @@ import sys
 from argparse import Namespace
 from typing import Union, Tuple, Literal
 
-import waybackpy
+import waybackpy  # type: ignore
 
 
 class SavepagenowFailureError(Exception):
@@ -10,15 +10,16 @@ class SavepagenowFailureError(Exception):
 
 
 class Archiver:
-
     @staticmethod
-    def from_parser_args(args: Namespace) -> 'Archiver':
+    def from_parser_args(args: Namespace) -> "Archiver":
         return Archiver(args.retry)
 
     def __init__(self, retry: int):
         """Init."""
         self.retry: int = retry
-        self.UA: str = "Mozilla/5.0 (Windows NT 5.1; rv:40.0) " "Gecko/20100101 Firefox/40.0"
+        self.UA: str = (
+            "Mozilla/5.0 (Windows NT 5.1; rv:40.0) " "Gecko/20100101 Firefox/40.0"
+        )
 
     def archive(self, url: str) -> Union[Literal[False], Tuple[str, bool]]:
         """Archive link."""

--- a/wbsv/crawler.py
+++ b/wbsv/crawler.py
@@ -3,7 +3,7 @@ from typing import List, Iterable, Set
 from urllib.parse import urldefrag, urljoin, urlparse
 
 import requests
-from bs4 import BeautifulSoup as BS
+from bs4 import BeautifulSoup as BS  # type: ignore
 
 from wbsv.url_filters import SchemaFilter, CombinedFilter, OwnDomainFilter, UrlFilter
 
@@ -14,32 +14,39 @@ class MissingURLSchemaWarning(UserWarning):
 
 class Crawler:
     @staticmethod
-    def from_parser_args(args: Namespace) -> 'Crawler':
+    def from_parser_args(args: Namespace) -> "Crawler":
         return Crawler.from_args(args.url, args.own, args.only_target, args.level)
 
     @staticmethod
-    def from_args(urls: Iterable[str], own: bool, only_target: bool, level: int) -> 'Crawler':
+    def from_args(
+        urls: Iterable[str], own: bool, only_target: bool, level: int
+    ) -> "Crawler":
         schema_filter = SchemaFilter()
         normalized_urls = Crawler._normalize_urls(urls, schema_filter)
-        domain_filters = [
-            schema_filter,
-            OwnDomainFilter.from_string_urls(set(normalized_urls))
-        ] if own else [schema_filter]
+        domain_filters = (
+            [schema_filter, OwnDomainFilter.from_string_urls(set(normalized_urls))]
+            if own
+            else [schema_filter]
+        )
         return Crawler(
             urls=normalized_urls,
             domain_filter=CombinedFilter(domain_filters),
             only_target=only_target,
-            level=level
+            level=level,
         )
 
-    def __init__(self, urls: List[str], domain_filter: UrlFilter, only_target: bool, level: int):
+    def __init__(
+        self, urls: List[str], domain_filter: UrlFilter, only_target: bool, level: int
+    ):
         """Init."""
         self.urls = urls
         self.domain_filter = domain_filter
         self.only_target = only_target
         self.level = level
         self.queue: List[Set[str]] = []
-        self.UA: str = "Mozilla/5.0 (Windows NT 5.1; rv:40.0) " "Gecko/20100101 Firefox/40.0"
+        self.UA: str = (
+            "Mozilla/5.0 (Windows NT 5.1; rv:40.0) " "Gecko/20100101 Firefox/40.0"
+        )
 
     def run_crawl(self) -> List[Set[str]]:
         """Execute crawler."""

--- a/wbsv/main.py
+++ b/wbsv/main.py
@@ -82,7 +82,7 @@ def parse_args(args_list: Optional[List[str]] = None) -> argparse.Namespace:
     parser.add_argument(
         "-O",
         "--own",
-        action='store_true',
+        action="store_true",
         help="Only URLs with the same domain as target",
     )
     parser.add_argument(
@@ -94,7 +94,9 @@ def parse_args(args_list: Optional[List[str]] = None) -> argparse.Namespace:
         return parser.parse_args()
 
 
-def wbsv(urls: Iterable[str], own: bool, only_target: bool, level: int, retry: int) -> None:
+def wbsv(
+    urls: Iterable[str], own: bool, only_target: bool, level: int, retry: int
+) -> None:
     past, now, fail = 0, 0, 0
     print("[+]Target: {}".format(urls))
     c = Crawler.from_args(urls=urls, own=own, only_target=only_target, level=level)
@@ -123,7 +125,9 @@ def wbsv_from_parser_args(args: argparse.Namespace) -> None:
     wbsv(args.url, args.own, args.only_target, args.level, args.retry)
 
 
-def cache_or_now(ind: int, len_links: int, archived_link: str, cached_flag: bool) -> Tuple[int, int]:
+def cache_or_now(
+    ind: int, len_links: int, archived_link: str, cached_flag: bool
+) -> Tuple[int, int]:
     if cached_flag:
         print("[{:02d}/{}]: <PAST> {}".format(ind, len_links, archived_link))
         return 1, 0
@@ -133,7 +137,7 @@ def cache_or_now(ind: int, len_links: int, archived_link: str, cached_flag: bool
 
 
 def wbsv_repl(args: argparse.Namespace) -> None:
-    finish_words = {"end", "exit", "exit()", "break", "bye", ":q", "finish"}
+    finish_words = ("end", "exit", "exit()", "break", "bye", ":q", "finish")
     print("[[Input a target url (ex: https://google.com)]]")
     while True:
         link = input(">>> ").rstrip()

--- a/wbsv/url_filters.py
+++ b/wbsv/url_filters.py
@@ -5,7 +5,6 @@ from urllib.parse import ParseResult, urlparse
 
 
 class UrlFilter(ABC):
-
     @abstractmethod
     def test_url(self, parsed_url: ParseResult) -> bool:
         pass
@@ -13,7 +12,9 @@ class UrlFilter(ABC):
 
 @dataclasses.dataclass
 class SchemaFilter(UrlFilter):
-    valid_schemas: Set[str] = dataclasses.field(default_factory=lambda: {"http", "https", "ftp", "file"})
+    valid_schemas: Set[str] = dataclasses.field(
+        default_factory=lambda: {"http", "https", "ftp", "file"}
+    )
 
     def test_url(self, parsed_url: ParseResult) -> bool:
         return parsed_url.scheme in self.valid_schemas
@@ -24,7 +25,7 @@ class OwnDomainFilter(UrlFilter):
     own_domains: Set[str]
 
     @staticmethod
-    def from_string_urls(urls: Iterable[str]) -> 'OwnDomainFilter':
+    def from_string_urls(urls: Iterable[str]) -> "OwnDomainFilter":
         return OwnDomainFilter({urlparse(url).netloc for url in urls})
 
     def test_url(self, parsed_url: ParseResult) -> bool:

